app/src/main/java/de/danoeh/antennapod/fragment/preferences/NextcloudGpodderPreferencesFragment.java

```diff
@@ -0,0 +1,133 @@
+package de.danoeh.antennapod.fragment.preferences;
+
+import android.app.Activity;
+import android.os.Bundle;
+import androidx.core.text.HtmlCompat;
+import androidx.preference.PreferenceFragmentCompat;
+
+import android.text.Spanned;
+import android.text.format.DateUtils;
+import com.google.android.material.snackbar.Snackbar;
+import com.nextcloud.android.sso.AccountImporter;
+import com.nextcloud.android.sso.exceptions.AndroidGetAccountsPermissionNotGranted;
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppAccountNotFoundException;
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppNotInstalledException;
+import com.nextcloud.android.sso.exceptions.NoCurrentAccountSelectedException;
+import com.nextcloud.android.sso.helper.SingleAccountHelper;
+import com.nextcloud.android.sso.model.SingleSignOnAccount;
+import com.nextcloud.android.sso.ui.UiExceptionManager;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.PreferenceActivity;
+import de.danoeh.antennapod.core.event.SyncServiceEvent;
+import de.danoeh.antennapod.core.preferences.GpodnetPreferences;
+import de.danoeh.antennapod.core.sync.SyncService;
+import de.danoeh.antennapod.dialog.AuthenticationDialog;
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+
+public class NextcloudGpodderPreferencesFragment extends PreferenceFragmentCompat {
```

Comment:
To be honest, I'm not 100% happy about the naming here. It feels a bit "duplicated" to have both gpodder and nextcloud in the title. Not sure what to use instead.

Reply:
What about NextcloudGPodderApi or GPodderNextcloudApp?

Reply:
Actually, if the nextcloud podcast app will get sync support, I would just call it only "NextcloudPreferencesFragment" without even mentioning gpodder. The internal fact that the sync service uses the same API doesn't need to be in the name.